### DISCUSSION TITLE
Connection closed by UNKNOWN port 65535

### DISCUSSION THREAD

**Discussion by Bannirui** [community/community#73011] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hi,

I get stuck in ssh proxy, appreciate for help.

### 1 OS platform
macOS 13.5
arch 6.4.10

### 2 corkscrew ver
2.0

### 3 ssh config

```shell
❯ cat .ssh/config

# my-github
Host github.com
	Hostname github.com
	IdentityFile ~/.ssh/my_github
	User bannirui@outlook.com
    ProxyCommand corkscrew 127.0.0.1 7890 %h %p
```
#### 4.1 ssh -T

```shell
❯ ssh -T git@github.com

kex_exchange_identification: Connection closed by remote host
Connection closed by UNKNOWN port 65535
```

#### 4.2 ssh -v

```shell
❯ ssh -v git@github.com

OpenSSH_9.0p1, LibreSSL 3.3.6
debug1: Reading configuration data /Users/dingrui/.ssh/config
debug1: /Users/dingrui/.ssh/config line 2: Applying options for github.com
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 21: include /etc/ssh/ssh_config.d/* matched no files
debug1: /etc/ssh/ssh_config line 54: Applying options for *
debug1: Authenticator provider $SSH_SK_PROVIDER did not resolve; disabling
debug1: Executing proxy command: exec corkscrew 127.0.0.1 7890 github.com 22
debug1: identity file /Users/dingrui/.ssh/my_github type 0
debug1: identity file /Users/dingrui/.ssh/my_github-cert type -1
debug1: Local version string SSH-2.0-OpenSSH_9.0
kex_exchange_identification: Connection closed by remote host
Connection closed by UNKNOWN port 65535
```

I hava no ideals on this, i'm looking forward to your suggestions or help, thanks again.

**Comment by ghostinhershell** [comment]
Hey there!  👋

Thanks for posting in the GitHub Community, @Bannirui! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category. The Accessibility category is a place for our community to discuss and provide feedback on the digital accessibility of GitHub products. Digital accessibility means that GitHub tools, and technologies, are designed and developed so that people with disabilities can use them.

With that being said, are you still seeing this issue on your end? On first look it looks like maybe you ran into an incident where services may be down. If you're not currently seeing this issue, but happen to run into it again, you can check the status of GitHub services at https://www.githubstatus.com/. 

I've gone ahead and moved this to the correct category for you. Good luck!

**Reply by Bannirui** [reply]
hi,

thank you for correcting the category. i have no relevant experience on github community, so i just put this question into a topic randomly.

then, i still have confuse on your words as below:

> On first look it looks like maybe you ran into an incident where services may be down

i do not get `service crash`, what does it mean, git server or corkscrew executor.

my ssh config is as below

```shell
Host github.com
        Hostname github.com
        Port 22
        IdentityFile ~/.ssh/my_github
        User Bannirui@outlook.com
        # some problem if using proxy
        # kex_exchange_identification: Connection closed by remote host
       ProxyCommand corkscrew 127.0.0.1 7890 %h %p

# zto-gitlab
Host git.ztosys.com
        Hostname git.ztosys.com
        IdentityFile ~/.ssh/zto_gitlab
        User dingrui0@zto.com
        ProxyCommand corkscrew 127.0.0.1 7890 %h %p
```

it throws exception as i mentationed since i run `ssh -T git@github.com`, but successfully for gitlab.

also, for github ssh, it works pretty good wihout proxy, but failed while using proxy.

so, i guess if github

**Reply by JounQin** [reply]
Any news on this issue?

```ssh
Host *
  AddKeysToAgent yes
  UseKeychain yes
  IdentityFile ~/.ssh/id_rsa
  TCPKeepAlive yes
  ProxyCommand socat - PROXY:127.0.0.1:%h:%p,proxyport=7890
```

```sh
ssh -v git@github.com
```

```log
OpenSSH_9.4p1, LibreSSL 3.3.6
debug1: Reading configuration data /Users/JounQin/.ssh/config
debug1: Reading configuration data /Users/JounQin/.ssh/code_gitpod.d/config
debug1: Reading configuration data /Users/JounQin/.orbstack/ssh/config
debug1: /Users/JounQin/.ssh/config line 8: Applying options for *
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 21: include /etc/ssh/ssh_config.d/* matched no files
debug1: /etc/ssh/ssh_config line 54: Applying options for *
debug1: Authenticator provider $SSH_SK_PROVIDER did not resolve; disabling
debug1: Executing proxy command: exec socat - PROXY:127.0.0.1:github.com:22,proxyport=7890
debug1: identity file /Users/JounQin/.ssh/id_rsa type 0
debug1: identity file /Users/JounQin/.ssh/id_rsa-cert type -1
debug1: Local version string SSH-2.0-OpenSSH_9.4
kex_exchange_identification: Connection closed by remote host
Connection closed by UNKNOWN port 65535
```

**Comment by Bannirui** [comment, marked as answer]
Hi there,

I have get done, and i recorded as below blog.

https://bannirui.github.io/2024/01/24/%E4%BB%A3%E7%90%86git%E7%9A%84ssh%E5%8D%8F%E8%AE%AE/

获取Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by cybershape** [reply]
Thanks a lot, you saved my day.

**Reply by juneleung** [reply]
thank you!

**Reply by S-Marouene** [reply]
it work fine ! thank you .

**Reply by tsingfei** [reply]
It works. Thank you!

**Reply by zhangyangeng** [reply]
thank you

**Reply by ningxiaowa** [reply]
Thank you!

**Reply by chansonzhang** [reply]
Thank you!

**Comment by goldpriceturkey5-max** [comment]
It looks like the connection is being closed before the key exchange starts, which usually points to an issue with the proxy rather than SSH itself. A couple of things you can try:

Test corkscrew directly
Run:

corkscrew 127.0.0.1 7890 github.com 22


If it hangs or fails, then the proxy isn’t forwarding properly.

Check if the proxy supports SSH
Some HTTP proxies don’t allow tunneling raw SSH traffic on port [22](https://goldpriceturkey.com/). If that’s the case, you can try:

Using ProxyJump (SSH over another server that has internet access).

Switching to connect instead of corkscrew (often more reliable).

Test without proxy
Just to confirm your key works:

ssh -i ~/.ssh/my_github -T git@github.com


If that succeeds, then the problem is definitely the proxy layer.

Alternative approach
GitHub supports SSH over HTTPS (port 443). You can try this instead in your config:

Host github.com
  Hostname ssh.github.com
  Port 443
  User git
  IdentityFile ~/.ssh/my_github
  ProxyCommand corkscrew 127.0.0.1 7890 %h %p


This forces SSH over port 443, which often works better through proxies.
